quantities/named_quantities.hpp

```diff
@@ -32,10 +35,16 @@ using SquareRoot = NthRoot<Q, 2>;
 template<typename Q>
 using CubeRoot = NthRoot<Q, 3>;
 
-// The result type of the derivative of a |Value|-valued function with respect
-// to its |Argument|-valued argument.
+// The result type of the N-th derivative of a |Value|-valued function with
+// respect to its |Argument|-valued argument.
+template<typename Value, typename Argument, int order>
```

Comment:
Could we have just one type alias with `int order=1`, calling it `Derivative`?

Reply:
Done, also for the plural form (`NthDerivatives`).